apps/total-typescript/src/purchase-transfer/purchase-transfer.tsx

```diff
@@ -0,0 +1,145 @@
+import {trpc} from '../trpc/trpc.client'
+import {useForm} from 'react-hook-form'
+import {PurchaseUserTransfer} from '@skillrecordings/database'
+import * as React from 'react'
+
+type PurchaseTransferFormData = {
+  email: string
+}
+
+const PurchaseTransferForm = ({
+  purchaseUserTransferId,
+  refetch,
+}: {
+  purchaseUserTransferId: string
+  refetch: () => Promise<void>
+}) => {
+  const utils = trpc.useContext()
+  const {
+    register,
+    handleSubmit,
+    watch,
+    reset,
+    formState: {errors},
+  } = useForm<PurchaseTransferFormData>()
+  const {data: purchaseUserTransfer, status} =
+    trpc.purchaseUserTransfer.byId.useQuery({
+      id: purchaseUserTransferId,
+    })
+
+  console.log({purchaseUserTransfer, status})
+
+  const {mutate, error, isLoading} =
+    trpc.purchaseUserTransfer.initiate.useMutation({
+      onSuccess: async (input) => {
+        utils.invalidate()
+        reset()
+        refetch()
+      },
+    })
+  const onSubmit = (data: PurchaseTransferFormData) => {
+    mutate({
+      purchaseUserTransferId: purchaseUserTransferId,
+      email: data.email,
+    })
+  }
+
+  return (
+    <div>
+      <form
+        className="flex w-full flex-col gap-2 text-left md:flex-row"
+        onSubmit={handleSubmit(onSubmit)}
+      >
```

Comment:
pretty awesome the way trpc trims down the logic/complexity in the component layer of this feature